### ISSUE TITLE
No logs are shown in flow run after upgrading from Prefect Server 2.8.5 to 2.10.8

### ISSUE THREAD

**Issue by filpano** [PrefectHQ/prefect#9532]
### First check

- [X] I added a descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the Prefect documentation for this issue.
- [X] I checked that this issue is related to Prefect and not one of its dependencies.

### Bug summary

I updated from Prefect `2.8.5` to `2.10.8` recently. I noticed that none of my flow runs have any logging shown anymore:

![image](https://github.com/PrefectHQ/prefect/assets/5376771/97182a5f-d007-4b55-b9f7-798cf65fc6df)

Since I can't imagine I'm the first person this has happened to with the latest version, I believe that some oddity may be causing it to happen.

Checking the Database `log` table, I do not see any new entries for my flow run. 

Is there any way to debug this behaviour? 

### Reproduction

```python3
I'm not sure what code to provide, as the error appeared after upgrading from Prefect Server `2.8.5` to `2.10.8`. The log output was present prior to the upgrade, and the `log` database table was being populated.
```

### Error

```python3
I do not see any errors in the Prefect Server or Agent logs.
```

### Versions

```Text
Version:             2.10.8
API version:         0.8.4
Python version:      3.10.6
Git commit:          79093235
Built:               Mon, May 8, 2023 12:23 PM
OS/Arch:             linux/x86_64
Profile:             default
Server type:         server
```

### Additional context

I do not have any custom logging configuration enabled. I did at one point (using a custom `logging.yml` and `PREFECT_LOGGING_SETTINGS_PATH`), but removed it after noting the error. The error persists.

**Comment by madkinsz** [comment]
Is this specific to deployed flow runs or any flow run? 

May be resolved by #9516

**Comment by filpano** [comment]
The behaviour seems to occur for flows executed via the CLI as well. Running a very trivial:

```
from prefect import flow

@flow
def run():
    pass


if __name__ == "__main__":
    run()
```

logs only the following:

```
(.venv) prefect@prefect:~$ python flow.py 
Overriding existing alarm handler <function _alarm_based_timeout.<locals>.sigalarm_to_error at 0x7f32a74c5cf0>
```

**Comment by madkinsz** [comment]
I think it's very likely that migrations have been done between those versions — you'd need to downgrade the database before downgrading the library and yeah you'd probably lose data related to any new tables.

What does `PREFECT_DEBUG_MODE=1 python -c "from prefect.logging import get_logger; get_logger().info('test')"` output?

**Comment by filpano** [comment]
Sorry for the late reply, a cold knocked me out this week.

I don't understand it at all, but a day after our last discussion in this thread, logs started appearing again in my flow runs (both scheduled deployment flow runs as well as manual flow runs).

As I mentioned we did use to have a `logging.yml` in `~/.prefect`, but this was the base [logging.yml](https://github.com/PrefectHQ/prefect/blob/orion/src/prefect/logging/logging.yml) with the additions for a file handler (following [this](https://discourse.prefect.io/t/how-to-stream-prefect-logs-to-a-file/1963) guide), and I'd removed the `logging.yml` when debugging the above issue.

I haven't changed anything since my last reply, and I'd restarted the prefect server and agents multiple times after removing the above config, so I'm really confused now. Is there perhaps some config being written to and then read from temporary files somewhere?

I can try replicating the steps I did with adding a file handler to the logging configuration and see if it occurs again... I'll post my results here.
